cpp/oneapi/dal/algo/kmeans/test/batch.cpp

```diff
@@ -499,4 +541,161 @@ TEMPLATE_LIST_TEST_M(kmeans_batch_test,
+
+    const auto first_row = row_accessor<const Float>(x).pull({ 0, 1 });
+    const auto c_init = homogen_table::wrap(first_row.get_data(), 1, column_count);
+
+    auto labels = array<std::int32_t>::zeros(1 * cluster_count);
+    auto label_ptr = labels.get_mutable_data();
+    auto first_label = &label_ptr[0];
+    std::iota(first_label, first_label + row_count, std::int32_t(0));
+    const auto y = homogen_table::wrap(labels.get_data(), row_count, 1);
+
+    this->exact_checks(x, x, x, y, 3, 1, 0.0);
+}
+
+TEMPLATE_LIST_TEST_M(kmeans_batch_test,
+                     "higgs/17/101",
+                     "[kmeans][nightly][batch][external-dataset]",
+                     kmeans_types) {
+    using Float = std::tuple_element_t<0, TestType>;
+
+    const te::dataframe data =
+        GENERATE_DATAFRAME(te::dataframe_builder{ "workloads/dataset/higgs/higgs_1m_test.csv" });
+    const table x = data.get_table(this->get_homogen_table_id());
+
+    constexpr std::int64_t cluster_count = 17;
+    constexpr std::int64_t max_iteration_count = 101;
+    constexpr Float ref_dbi = 0.0;
+    constexpr Float ref_obj_func = 0.0;
+
+    this->dbi_determenistic_checks(x,
+                                   cluster_count,
+                                   max_iteration_count,
+                                   0.0,
+                                   ref_dbi,
+                                   ref_obj_func);
+}
+
+TEMPLATE_LIST_TEST_M(kmeans_batch_test,
+                     "susy/250/10",
+                     "[kmeans][nightly][batch][external-dataset]",
+                     kmeans_types) {
+    using Float = std::tuple_element_t<0, TestType>;
+
+    const te::dataframe data =
+        GENERATE_DATAFRAME(te::dataframe_builder{ "workloads/susy/dataset/susy_test.csv" });
+    const table x = data.get_table(this->get_homogen_table_id());
+
+    constexpr std::int64_t cluster_count = 250;
+    constexpr std::int64_t max_iteration_count = 10;
+    constexpr Float ref_dbi = 0.0;
+    constexpr Float ref_obj_func = 0.0;
+
+    this->dbi_determenistic_checks(x,
+                                   cluster_count,
+                                   max_iteration_count,
+                                   0.0,
+                                   ref_dbi,
+                                   ref_obj_func);
+}
+
+TEMPLATE_LIST_TEST_M(kmeans_batch_test,
+                     "road_network_20t/111/13",
+                     "[kmeans][nightly][batch][external-dataset]",
+                     kmeans_types) {
+    using Float = std::tuple_element_t<0, TestType>;
+
+    const te::dataframe data = GENERATE_DATAFRAME(
+        te::dataframe_builder{ "workloads/dataset/road_network/road_network_100t_cluster.csv" });
+    const table x = data.get_table(this->get_homogen_table_id());
+
+    constexpr std::int64_t cluster_count = 111;
+    constexpr std::int64_t max_iteration_count = 13;
+    constexpr Float ref_dbi = 0.0;
+    constexpr Float ref_obj_func = 0.0;
```

Comment:
Are these metrics always zero for road net?

Reply:
This part is in progress for now